src/main/java/org/folio/coursereserves/util/CRUtil.java

```diff
@@ -515,202 +416,153 @@ public static Future<JsonObject> lookupItemByBarcode(String barcode,
+      Future<Term> termFuture;
+      Future<CourseType> coursetypeFuture;
+      Future<JsonObject> locationFuture;
+      Future<JsonObject> servicePointFuture;
+      if (expandTerm && termId != null) {
+        termFuture = lookupTerm(termId, okapiHeaders, context);
       } else {
-        try {
-          CourseListing courselisting = clRes.result();
-          String termId = courselisting.getTermId();
-          String courseTypeId = courselisting.getCourseTypeId();
-          String locationId = courselisting.getLocationId();
-          String servicepointId = courselisting.getServicepointId();
-          Future<Term> termFuture;
-          Future<CourseType> coursetypeFuture;
-          Future<JsonObject> locationFuture;
-          Future<JsonObject> servicePointFuture;
-          if(expandTerm && termId != null) {
-            termFuture = lookupTerm(termId, okapiHeaders, context);
-          } else {
-            termFuture = Future.failedFuture("No lookup");
-          }
-          if(expandTerm && courseTypeId != null) {
-            coursetypeFuture = lookupCourseType(courseTypeId, okapiHeaders, context);
-          } else {
-            coursetypeFuture = Future.failedFuture("No lookup");
-          }
-          if(expandTerm && locationId != null) {
-            locationFuture = lookupLocation(locationId, okapiHeaders, context);
-          } else {
-            locationFuture = Future.failedFuture("No lookup");
-          }
-          if(expandTerm && servicepointId != null) {
-            servicePointFuture = lookupServicepoint(servicepointId, okapiHeaders, context);
-          } else {
-            servicePointFuture = Future.failedFuture("No lookup");
-          }
-
-          List<Future> futureList = new ArrayList<>();
-          futureList.add(termFuture);
-          futureList.add(coursetypeFuture);
-          futureList.add(locationFuture);
-          futureList.add(servicePointFuture);
-          CompositeFuture compositeFuture = CompositeFuture.join(futureList);
-          compositeFuture.onComplete(compRes -> {
-            try {
-              if(termFuture.succeeded()) {
-                courselisting.setTermObject(termObjectFromTerm(termFuture.result()));
-              }
-              if(coursetypeFuture.succeeded()) {
-                courselisting.setCourseTypeObject(courseTypeObjectFromCourseType(
-                    coursetypeFuture.result()));
-              }
-              if(locationFuture.succeeded()) {
-                courselisting.setLocationObject(locationObjectFromJson(locationFuture.result()));
-              }
-              if(servicePointFuture.succeeded()) {
-                courselisting.setServicepointObject(servicepointObjectFromJson(servicePointFuture.result()));
-              }
-              promise.complete(courselisting);
-            } catch(Exception e) {
-              promise.fail(e);
+        termFuture = Future.failedFuture("No lookup");
+      }
```

Comment:
Moving related code to one place:
```
      List<Future> futureList = new ArrayList<>();
      if (termId != null) {
        futureList.add(
            lookupTerm(termId, okapiHeaders, context)
            .map(CRUtil::termObjectFromTerm)
            .onSuccess(courselisting::setTermObject)
        );
      }
      if (courseTypeId != null) {
        futureList.add(
          lookupCourseType(courseTypeId, okapiHeaders, context)
          .map(CRUtil::courseTypeObjectFromCourseType)
          .onSuccess(courselisting::setCourseTypeObject)
      )
      ...
      return CompositeFuture.join(futureList)
          .recover(x -> Future.succeededFuture())  // is this still needed?
          .map(courselisting);
```

Reply:
Good idea. Will do (probably not today).